docs/proposal/20210311-single-controller-multitenancy.md

```diff
@@ -0,0 +1,298 @@
+1. To maintain backwards compatibility for users who do not intend to use these capabilities and want to continue specifying credentials through the CAPV deployment
+
+## Proposal
+
+### User Stories
+
+#### Story 1 - Deploying to multiple vCenters
+
+Alex is an engineer in an organization that enforces strict vSphere account and environment architectures. They use a management vCenter and account for the management cluster where CAPV is running. Alex has a workload account in a dedicated vCenter.
+Alex can provision a new cluster in the workload account by creating Cluster API resources in the management cluster. The CAPV controller will utilize the workload credentials to provision the cluster in Alex's environment.
+
+#### Story 2 - Legacy behaviour
+
+Erin is an engineer in a smaller less strict organization. They use a single vCenter and keep their management cluster up-to-date. Erin can create new vSphere clusters while omitting the `vSphereCluster.Spec.AccountRef` field. The CAPV controller will use the credentials it was given at initialization to create new vSphere clusters.
+
+### Requirements
+
+#### Functional Requirements
+
+##### FR1. CAPV MUST support credentials provided through a `VsphereAccount` and referenced by `VsphereCluster.AccountRef` field
+
+##### FR2. CAPV MUST support static credentials
+
+##### FR3. CAPV MUST support clusterctl move scenarios
+
+##### FR4. CAPV MUST prevent privilege escalation allowing users to create clusters in accounts they should not be able to
+
+#### Non-Functional Requirements
+
+##### NFR1. Unit tests MUST exist for cluster and machine controllers that utilize the credentials
+
+##### NFR2. e2e tests MUST exist for multi account scenarios
+
+### Implementation Details/Notes/Constraints
+
+#### Current State
+
+CAPV currently uses a session manager to create and cache sessions. Sessions are created (or retrieved from the cache) during cluster and vsphere VM reconcile loops.
+vSphere cluster uses the session as a sanity check to ensure connectivity. vSphere VM uses the session for VM lifecycle tasks on vcenter and the session is stored in a VMContext.
+
+```go
+type VMContext struct {
+    *ControllerContext
+    VSphereVM   *v1alpha3.VSphereVM
+    PatchHelper *patch.Helper
+    Logger      logr.Logger
+    Session     *session.Session
+}
+```
+
+#### Proposed Changes
+
+The proposed changes below allow users to specify the account to use for clusters during runtime.
+
+Changed Resources
+
+- `VSphereCluster`
+
+New Resources
+
+- cluster-scoped `VSphereAccount` represents the vCenter account to use for reconciliation. This type should also contain a string list of namespaces in which it is allowed to be used.
+
+Changes to VSphereCluster
+A new field is added to the VSphereClusterSpec to reference the `VSphereAccount`. We intend to use `corev1.LocalObjectReference` in order to ensure the only objects that can be referenced are either in the same namespace or are scoped globally.
```

Comment:
We've created our own PrincipalReference type in CAPA rather than reuse TypedLocalObjectReference.